portfolio/src/main/webapp/calendarApi.js

```diff
@@ -0,0 +1,283 @@
+
+            if (!doesEventExist) {
+              updateCalendarEventList(newCalendarEvent);
+            }
+          });
+        }
+      });
+}
+
+/**
+ * Retrieves the logged in user's Google email and uses that email
+ * to embed the user's Google Calendar on the UI.
+ * The calendar is displayed in the user's Google Calendar's time zone.
+ */
+function showCalendarView(user) {
+  const userEmail = encodeURIComponent(user.getBasicProfile().getEmail());
+
+  // Fetches the user's Calendar's time zone.
+  gapi.client.calendar.settings.get({setting: 'timezone'})
+      .then((responseTimeZone) => {
+        const userCalendarTimeZone =
+            encodeURIComponent(responseTimeZone.result.value);
+        const calendarViewUrl =
+            'https://calendar.google.com/calendar/embed?src=' + userEmail +
+            '&ctz=' + userCalendarTimeZone + '&mode=WEEK';
+
+        // Updates the UI so that calendar view appears.
+        $('#calendar-view').removeClass('d-none').attr('src', calendarViewUrl);
+      });
+}
+
+/**
+ * Onclick function for 'Looks good to me, export" button.
+ * Asks the user for Write access to the API scope.
+ */
+function addWriteScope() {
+  var GoogleUser = GoogleAuth.currentUser.get();
+  GoogleUser.grant({scope: SCOPE_READ_WRITE})
+      .then((response) => {
+        $('#export-calendar-message').addClass('d-none');
+        addNewEventsToGoogleCalendar();
+      })
+      .catch(function(error) {
+        handleExportAuthError(error);
+      })
+}
+
+/**
+ * Adds the scheduled task items back to the user's Google Calendar.
+ * TODO(hollyyuqizheng): fill in the rest of the method once
+ * results can be returned from the scheduling algorithm.
+ */
+function addNewEventsToGoogleCalendar() {
+  const event = {};
+  addOneEventToGoogleCalendar(event);
+}
+
+/** Adds an individual event to the authorized user's Google Calendar. */
+function addOneEventToGoogleCalendar(event) {
+  const request = gapi.client.calendar.events.insert(
+      {calendarId: 'primary', resource: event});
+
+  request.execute(function() {
+    // Refreshes the calendar view so that the new event shows up on it.
+    showCalendarView(GoogleAuth.currentUser.get());
+  });
+}
+
+/**
+ * Updates import message box based on the error during authentication process
+ * for exporting
+ */
+function handleExportAuthError(e) {
+  var $exportCalendarMessage = $('#export-calendar-message');
+
+  var errorMessage;
+  if (e.error === ERROR_CODES.popup_closed_by_user) {
+    errorMessage =
+        'It seems like you didn\'t complete the authorization process. ' +
+        'Please click the Export button again.';
+  } else if (e.error === ERROR_CODES.access_denied) {
+    errorMessage =
+        'You didn\'t give permission to update your Google Calendar, ' +
+        'so your task schedule cannot be exported.'
+  } else {
+    errorMessage = 'An error occurred';
+  }
+
+  $exportCalendarMessage.removeClass('d-none').text(errorMessage);
```

Comment:
nit: I think it might make more sense to call text(...) and then call removeClass(...)

Reply:
Updated.